node/connection_base.js

```diff
@@ -146,14 +146,22 @@ function buildResponse(req, options) {
     var self = this;
 
     if (req.res && req.res.state !== States.Initial) {
-        self.errorEvent.emit(self, errors.ResponseAlreadyStarted({
+        req.errorEvent.emit(req, errors.ResponseAlreadyStarted({
```

Comment:
Plus, having to test this pointed out that this is the better errorEvent rather than on connection itself.